crowbar_framework/app/models/dns_service.rb

```diff
@@ -91,38 +96,45 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     return if all_nodes.empty?
 
     tnodes = role.override_attributes["dns"]["elements"]["dns-server"]
-    nodes = tnodes.map {|n| NodeObject.find_node_by_name n}
-
-    # electing master dns-server
-    master = nil
-    admin = nil
-    nodes.each do |node|
-      if node[:dns][:master]
-        master = node
-        break
-      elsif node.admin?
-        admin = node
+
+    if !tnodes.blank?
```

Comment:
Maybe this will save us from additional indentation
```
tnodes = role.override_attributes["dns"]["elements"]["dns-server"]
return if tnodes.blank?
```